### DISCUSSION TITLE
IS IBuffer supported at all in CsWinRT?

### DISCUSSION THREAD

**Discussion by pedasmith** [microsoft/CsWinRT#1207] [Q&A]
I've already got a big chunk of code written for UWP; it's intended to connect to Bluetooth devices and, as part of that, provides interfaces that use the IBuffer interface that UWP uses everywhere. I'm trying to use this code, unchanged, in a .NET Core 6 command line app.

When I try to make a .NET Core (6) command-line program that re-uses these UWP classes (via a "shared code" project, so I'm literally just sucking these .CS files into my project), I get tons of errors that "Error	CS0012	The type 'IBuffer' is defined in an assembly that is not referenced. You must add a reference to assembly 'Windows, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime'."

It's frankly driving me nuts. There's ton's of blog posts and stack overflow comments, but each and every one is not applicable: they suggest I do things that are obsolete since .NET 5, or only for the framework, or are incompatible.

Help!

**Comment by manodasanW** [comment]
@pedasmith IBuffer is indeed still supported.  What is the TFM for your .NET 6 command line program?  Are you targeting a Windows OS version-specific TFM to get access to the Windows WinRT API surface and thereby IBuffer as discussed in step 2 of [here](https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-enhance#net-6-and-later-use-the-target-framework-moniker-option)?

**Reply by pedasmith** [reply]
Yes, those are the steps I'm following.
```
My CSPROJ looks like this:
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
	  <CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>disable</Nullable>
  </PropertyGroup>
```

**Reply by manodasanW** [reply]
@pedasmith, thanks for confirming.  Are you able to provide a repro to demonstrate the issue you are seeing?

**Reply by pedasmith** [reply]
**Found the problem and the solution!**  What's going on is that over the years you've built a bazillion ways to interop between .NET Core and UWP, but without any logical way to decide which one to try first. It turns out that if you install some of the Nuget ways to interop, that interferes with this new thing.
